components/script/dom/imagedata.rs

```diff
@@ -180,18 +176,20 @@ impl ImageData {
 }
 
 impl ImageDataMethods for ImageData {
-    // https://html.spec.whatwg.org/multipage/#dom-imagedata-width
+    /// <https://html.spec.whatwg.org/multipage/#dom-imagedata-width>
     fn Width(&self) -> u32 {
         self.width
     }
 
-    // https://html.spec.whatwg.org/multipage/#dom-imagedata-height
+    /// <https://html.spec.whatwg.org/multipage/#dom-imagedata-height>
     fn Height(&self) -> u32 {
         self.height
     }
 
-    // https://html.spec.whatwg.org/multipage/#dom-imagedata-data
-    fn Data(&self, _: JSContext) -> NonNull<JSObject> {
-        NonNull::new(self.data.get()).expect("got a null pointer")
+    /// <https://html.spec.whatwg.org/multipage/#dom-imagedata-data>
+    fn Data(&self, _: JSContext) -> Uint8ClampedArray {
+        self.data
+            .get_internal()
+            .expect("Failed to get Data from ImageData.")
```

Comment:
Let's return `JsError` as we did in the other PR.

Reply:
should we do it for other changes we merged, we have lot of `.get_internal().expect` ?

Reply:
I think so, but it depends on the spec. It would be good to do this in a separate PR so each cases can be investigated separately(it may require updating test expectations also).

Reply:
I think we only return error when function returns `Fallible` but not in this case

Reply:
yes! Although we can check the spec to see if the function should be fallible...

Reply:
The spec doesn't say it direclty, for wording like "initialize the [data](https://html.spec.whatwg.org/multipage/canvas.html#dom-imagedata-data) attribute of imageData to a new [Uint8ClampedArray](https://webidl.spec.whatwg.org/#idl-Uint8ClampedArray) object" implies using the Web IDL and JS operations which are in fact fallible. 

Let's return JsError.

Reply:
done